include/grpcpp/test/default_reactor_test_peer.h

```diff
@@ -0,0 +1,50 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_TEST_DEFAULT_REACTOR_TEST_PEER_H
+#define GRPCPP_TEST_DEFAULT_REACTOR_TEST_PEER_H
+
+#include <grpcpp/server_context.h>
+
+namespace grpc {
+namespace testing {
+
+/// A test-only class to access private members and methods of ServerContext.
```

Comment:
I have expanded this comment. I'm a little loath to add more examples here because that will be covered separately in the codelab and also because we don't encourage mocking (it's a necessary evil for our broader codebase though).